ggml-threading.h

```diff
@@ -0,0 +1,120 @@
+#include <stdint.h>
+#include <llama.h>
+
+#if defined(_WIN32)
+
+#define WIN32_LEAN_AND_MEAN
+#ifndef NOMINMAX
+#define NOMINMAX
+#endif
+#include <windows.h>
+
+#ifdef  __cplusplus
+extern "C" {
+#endif
+
+typedef volatile LONG atomic_int;
+typedef atomic_int atomic_bool;
+
+extern void atomic_store(atomic_int* ptr, LONG val);
+extern LONG atomic_load(atomic_int* ptr);
+extern LONG atomic_fetch_add(atomic_int* ptr, LONG inc);
+extern LONG atomic_fetch_sub(atomic_int* ptr, LONG dec);
```

Comment:
I think it is very important that these functions are inlined, so they cannot be made `extern`. Additionally, we cannot export symbols with these names, because they are very likely to create conflicts. Same with the pthread and sched_yield stuff.

Reply:
I see.  Valid points.

That's the exact kind of feedback I was looking for.  Thanks.